src/platform/packages/shared/kbn-esql-ast/src/commands_registry/commands/rerank/autocomplete.ts

```diff
@@ -19,18 +39,201 @@ export async function autocomplete(
+
+      return getCommandMapExpressionSuggestions(innerText, availableParameters);
+    }
+
+    case CaretPosition.AFTER_COMMAND: {
+      return [pipeCompleteItem];
+    }
+
+    default: {
+      return [];
+    }
+  }
+}
+
+// ============================================================================
+// Suggestion Handlers
+// ============================================================================
+
+async function handleOnFieldList({
+  innerText,
+  callbacks,
+  context,
+  rerankCommand,
+}: {
+  innerText: string;
+  callbacks: ICommandCallbacks;
+  context: ICommandContext | undefined;
+  rerankCommand: ESQLAstRerankCommand;
+}): Promise<ISuggestionItem[]> {
+  const fieldSuggestions = (await callbacks.getByType?.(FIELD_LIST_TYPES)) ?? [];
+
+  const suggestions = await handleFragment(
+    innerText,
+    (fragment) => columnExists(fragment, context),
+    // incomplete: get available fields suggestions
+    (_: string, rangeToReplace?: { start: number; end: number }) => {
+      const customFieldSuggestion = getNewUserDefinedColumnSuggestion(
+        callbacks.getSuggestedUserDefinedColumnName?.() || ''
+      );
+
+      return [customFieldSuggestion, ...fieldSuggestions].map((suggestion) => ({
+        ...suggestion,
+        rangeToReplace,
+        command: TRIGGER_SUGGESTION_COMMAND,
+      }));
+    },
+    // complete: get next actions suggestions for completed field
+    (fragment: string, rangeToReplace: { start: number; end: number }) => {
+      const results = buildNextActions({ withSpaces: true });
+
+      return results.map((suggestion) => ({
+        ...suggestion,
+        filterText: fragment,
+        text: fragment + suggestion.text,
+        rangeToReplace,
+      }));
+    }
+  );
+
+  return suggestions;
+}
+
+async function handleOnExpression({
+  innerText,
+  callbacks,
+  context,
+  expressionRoot,
+}: {
+  innerText: string;
+  callbacks: ICommandCallbacks;
+  context: ICommandContext | undefined;
+  expressionRoot: ESQLSingleAstItem | undefined;
+}): Promise<ISuggestionItem[]> {
+  let suggestions = await suggestForExpression({
+    innerText,
+    getColumnsByType: callbacks.getByType,
+    expressionRoot,
+    location: Location.RERANK,
+    preferredExpressionType: 'boolean',
+    context,
+    hasMinimumLicenseRequired: callbacks.hasMinimumLicenseRequired,
+    activeProduct: context?.activeProduct,
+  });
 
-  // TODO: Add more specific autocomplete logic for RERANK command
-  const fieldSuggestions = await callbacks.getByType('any');
-  suggestions.push(...fieldSuggestions);
+  if (expressionRoot) {
+    const expressionType = getExpressionType(expressionRoot, context?.columns);
 
-  // Add the pipe completion item when appropriate
-  suggestions.push(pipeCompleteItem);
+    if (expressionType === 'boolean' && isExpressionComplete(expressionType, innerText)) {
+      const allowed = new Set(['AND', 'OR']); // TODO: this filter should be unnecessary. Need to fix suggestForExpression
```

Comment:
I will open an issue also for this situation.

In suggestExpression when you enter into the complete IF branch, we should only retrieve AND and OR operators. Instead we also see other operators.  This is a genera behaviour